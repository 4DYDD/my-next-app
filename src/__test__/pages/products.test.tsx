import ProductsPage from "@/pages/products";
import { render, screen } from "@testing-library/react";

jest.mock("next/router", () => {
  return {
    useRouter() {
      return {
        route: "/products",
        pathname: "",
        query: "",
        asPath: "",
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
        isReady: true,
      };
    },
  };
});

describe("Products Page", () => {
  it("should render Products Page", () => {
    const page = render(<ProductsPage />);
    expect(page).toMatchSnapshot();
  });
});
