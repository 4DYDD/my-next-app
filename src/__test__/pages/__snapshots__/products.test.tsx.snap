// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Page should render Products Page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="font-bold text-3xl mb-5"
      >
        Products
      </h1>
      <div
        class="container grid md:grid-cols-2 w-[95%] xl:grid-cols-3 overflow-y-auto h-[65vh] scrollbar-custom"
      >
        <ul
          class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
        >
          <li
            class=" flexc !justify-start w-full mb-5"
          >
            <div
              class="flexc w-full overflow-hidden rounded-xl"
            >
              <div
                class="bg-gray-400 animate-pulse size-72 xl:size-80"
              />
            </div>
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
        </ul>
        <ul
          class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
        >
          <li
            class=" flexc !justify-start w-full mb-5"
          >
            <div
              class="flexc w-full overflow-hidden rounded-xl"
            >
              <div
                class="bg-gray-400 animate-pulse size-72 xl:size-80"
              />
            </div>
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
        </ul>
        <ul
          class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
        >
          <li
            class=" flexc !justify-start w-full mb-5"
          >
            <div
              class="flexc w-full overflow-hidden rounded-xl"
            >
              <div
                class="bg-gray-400 animate-pulse size-72 xl:size-80"
              />
            </div>
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
        </ul>
        <ul
          class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
        >
          <li
            class=" flexc !justify-start w-full mb-5"
          >
            <div
              class="flexc w-full overflow-hidden rounded-xl"
            >
              <div
                class="bg-gray-400 animate-pulse size-72 xl:size-80"
              />
            </div>
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
        </ul>
        <ul
          class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
        >
          <li
            class=" flexc !justify-start w-full mb-5"
          >
            <div
              class="flexc w-full overflow-hidden rounded-xl"
            >
              <div
                class="bg-gray-400 animate-pulse size-72 xl:size-80"
              />
            </div>
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
          <li
            class="flexc !justify-start w-full my-0.5"
          >
            <span
              class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
            />
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="font-bold text-3xl mb-5"
    >
      Products
    </h1>
    <div
      class="container grid md:grid-cols-2 w-[95%] xl:grid-cols-3 overflow-y-auto h-[65vh] scrollbar-custom"
    >
      <ul
        class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
      >
        <li
          class=" flexc !justify-start w-full mb-5"
        >
          <div
            class="flexc w-full overflow-hidden rounded-xl"
          >
            <div
              class="bg-gray-400 animate-pulse size-72 xl:size-80"
            />
          </div>
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
      </ul>
      <ul
        class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
      >
        <li
          class=" flexc !justify-start w-full mb-5"
        >
          <div
            class="flexc w-full overflow-hidden rounded-xl"
          >
            <div
              class="bg-gray-400 animate-pulse size-72 xl:size-80"
            />
          </div>
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
      </ul>
      <ul
        class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
      >
        <li
          class=" flexc !justify-start w-full mb-5"
        >
          <div
            class="flexc w-full overflow-hidden rounded-xl"
          >
            <div
              class="bg-gray-400 animate-pulse size-72 xl:size-80"
            />
          </div>
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
      </ul>
      <ul
        class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
      >
        <li
          class=" flexc !justify-start w-full mb-5"
        >
          <div
            class="flexc w-full overflow-hidden rounded-xl"
          >
            <div
              class="bg-gray-400 animate-pulse size-72 xl:size-80"
            />
          </div>
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
      </ul>
      <ul
        class="m-3 p-5 flex-col flexc gap-1 rounded-lg shadow shadow-gray-300 border border-gray-300"
      >
        <li
          class=" flexc !justify-start w-full mb-5"
        >
          <div
            class="flexc w-full overflow-hidden rounded-xl"
          >
            <div
              class="bg-gray-400 animate-pulse size-72 xl:size-80"
            />
          </div>
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
        <li
          class="flexc !justify-start w-full my-0.5"
        >
          <span
            class="flexc !justify-start w-full animate-pulse bg-gray-400 h-6"
          />
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
